airflow/dag_processing/bundles/manager.py

```diff
@@ -90,7 +94,22 @@ def get_bundle(self, name: str, version: str | None = None) -> BaseDagBundle:
 
         :return: The DAG bundle.
         """
-        # TODO: proper validation of the bundle configuration so we have better error messages
-        bundle_config = self.bundle_configs[name]
-        bundle_class = import_string(bundle_config["classpath"])
-        return bundle_class(name=name, version=version, **bundle_config["kwargs"])
+        # todo (AIP-66): proper validation of the bundle configuration so we have better error messages
+        cfg_tuple = _bundle_config.get(name)
+        if not cfg_tuple:
+            self.parse_config()
+            cfg_tuple = _bundle_config.get(name)
+            if not cfg_tuple:
+                raise ValueError(f"Requested bundle '{name}' is not configured.")
```

Comment:
```suggestion
        if not _bundle_config():
            self.parse_config()
        
        cfg_tuple = _bundle_config.get(name)
        if not cfg_tuple:
            raise ValueError(f"Requested bundle '{name}' is not configured.")
```

nit: little nicer. can probably do that conditional check everywhere, or just call it everywhere and short circuit in `parse_config`.

Reply:
The thing is, you are calling `parse_config` unnecessarily when the bundle is already there.  There's a bunch of logic in parse config and it's sorta wasteful to call it when the config you are looking for is already there.  That's partly the point of caching the configured bundles i think.